### DISCUSSION TITLE
[remark-breaks] Can I ignore processing for code block?

### DISCUSSION THREAD

**Discussion by rikusen0335** [remarkjs/remark#889] [Q&A] [answered]
The code below (code1) is showing like code2 after remark is proceeded.
I'm using `remark-prism` to highlighting, `remark-breaks` to break in article.

code1
![image](https://user-images.githubusercontent.com/19174234/140973009-468355f0-4087-4572-b3ec-e52de2ecbfd7.png)

code2
![image](https://user-images.githubusercontent.com/19174234/140973087-7f158962-ddf1-41d0-a9da-b26dea0cc14e.png)

So is there way to ignore when process coe block, or just use some escape letters?
Thanks!

**Comment by ChristianMurphy** [comment, marked as answer]
@rikusen0335 could you share a runnable example of the issue you are seeing?
I tried a version of this `remark-breaks` with `@mapbox/rehype-prism` and am not seeing the issue https://codesandbox.io/s/remark-breaks-and-prism-kqtoy

In general, using a rehype plugin for highlighting will make things easier to handle (as most highlighters work directly with HTML rather than on markdown).

**Reply by rikusen0335** [reply]
Thank you so much for answering!
I'm not really sure so I couldn't share you a runnable example, but changing `remark-prism` to `@mapbox/rehype-prism` seems to solve the issue!

Also, I am using them through the `next-mdx-remote` lib to proceed it and it is now working perfectly.
The code final is look like this:
```js
  const content = await serialize(post.content, {
    mdxOptions: {
      remarkPlugins: [
        remarkHint,
        remarkBreaks,
      ],
      rehypePlugins: [
        rehypePrism,
      ]
    },
  })
```

Anyway I am not sure what does differnence and which/when to use remark plugin and rehype plugin. If you have any good please provide me them (sorry even this is discussion Q&A)

**Reply by ChristianMurphy** [reply]
> Anyway I am not sure what does difference and which/when to use remark plugin and rehype plugin. If you have any good please provide me them

Some context, remark and mdast work specifically on markdown and don't know about HTML or other formats, rehype and hast work specifically on HTML or hyperscript (JSX) and don't know about markdown or other formats. remark-rehype bridges markdown/mdast to html/hast and translates between the two.

A rule of thumb knowing which type of plugin to use:

remark plugins are generally helpful if:
* you want to validate markdown
* you want to add new syntax to markdown, like tables, footnotes, or math
* a library being wrapped with unified specifically expects/works on markdown
* you want to write back to markdown

rehype plugins are generally helpful if:
* you want to change how the HTML or JSX output is generated
* a library being wrapped with unified specifically expects/works on HTML

---

In the case of Prism, it doesn't necessarily change anything in the markdown itself, it changes the HTML/JSX generated so a rehype plugin is most effective at helping.

**Reply by rikusen0335** [reply]
Ohh thanks for really clear information, I finally understood. Really appreciate it!
